crates/core/src/backend/decrypt.rs

```diff
@@ -356,8 +361,19 @@ impl<C: CryptoKey> DecryptBackend<C> {
             be,
             key,
             zstd: None,
+            extra_check: false,
         }
     }
+
+    fn decrypt_file(&self, data: &[u8]) -> RusticResult<Vec<u8>> {
+        let decrypted = self.decrypt(data)?;
+        Ok(match decrypted.first() {
+            Some(b'{' | b'[') => decrypted, // not compressed
```

Comment:
Would be determining compression by magic number (zstd is `0xFD2FB528` i think?) better here?

From this issue: https://github.com/facebook/zstd/issues/768

Reply:
No - this is implemented here as defined in the restic repo specification.